grammar.js

```diff
@@ -1035,24 +1044,30 @@ module.exports = grammar({
 
     symbol_literal: $ => '__no_longer_used',
 
-    _plainid: $ => /[a-zA-Z_\\$][\w\\$]*/,
+    /**
+     * id               ::=  plainid
+     *                       |  ‘`’ { charNoBackQuoteOrNewline | UnicodeEscape | charEscapeSeq 
+     */
+    identifier: $ => prec.left(choice(
+      $._alpha_identifier,
+      $._backquoted_id,
+    )),
+
+    /**
+     * alphaid          ::=  upper idrest
+     *                       |  varid
+     *
+     * The following regex was copied from Ethan's https://github.com/eatkins/tree-sitter-ebnf-generator/blob/b25508d4ed3c6f9926acce70b2a1d0847d45363a/examples/scala/grammar.js#L40-L41 minus the operator part, which we handle using operator_identifier.
+     */
+    _alpha_identifier: $ => /[\p{Lu}\p{Lt}\p{Nl}\p{Lo}\p{Lm}\$][\p{Lu}\p{Lt}\p{Nl}\p{Lo}\p{Lm}\$\p{Ll}_\u00AA\u00BB\u02B0-\u02B8\u02C0-\u02C1\u02E0-\u02E4\u037A\u1D78\u1D9B-\u1DBF\u2071\u207F\u2090-\u209C\u2C7C-\u2C7D\uA69C-\uA69D\uA770\uA7F8-\uA7F9\uAB5C-\uAB5F0-9]*(_[!#%&*+-\/:<=>?@'^\|‘~]+)?|[\p{Ll}_\u00AA\u00BB\u02B0-\u02B8\u02C0-\u02C1\u02E0-\u02E4\u037A\u1D78\u1D9B-\u1DBF\u2071\u207F\u2090-\u209C\u2C7C-\u2C7D\uA69C-\uA69D\uA770\uA7F8-\uA7F9\uAB5C-\uAB5F_][\p{Lu}\p{Lt}\p{Nl}\p{Lo}\p{Lm}\$\p{Ll}_\u00AA\u00BB\u02B0-\u02B8\u02C0-\u02C1\u02E0-\u02E4\u037A\u1D78\u1D9B-\u1DBF\u2071\u207F\u2090-\u209C\u2C7C-\u2C7D\uA69C-\uA69D\uA770\uA7F8-\uA7F9\uAB5C-\uAB5F0-9]*(_[!#%&*+-\/:<=>?@'^\|‘~]+)?/,
```

Comment:
> (_[!#%&*+-\/:<=>?@'^\|‘~]+)

I can see the regex still contains the operator part - is the comment accurate then?

Reply:
Yea. Note that symbolic characters follow mandatory underscore, e.g. for `unary_!`.